enhancements/cluster-api/installing-cluster-api-components-in-ocp.md

```diff
@@ -0,0 +1,290 @@
+1. Once the main branch has been rebased locally, the Rebase Bot invokes the relevant Lifecycle Hook (e.g. post-rebase) an executes the corresponding command defined for this stage, set to be the manifests generation tool (which resides in the [Cluster CAPI Operator](https://github.com/openshift/cluster-capi-operator) repository, and it is imported in the provider as a tool),
+which generates the manifests for the CAPI Provider (Deployments, RBACs, CRDs), it embeds them into the transport ConfigMaps, and dumps them to the `/openshift/manifests` folder in the Provider source tree. **TODO(@damdo): describe whether the CRDs/RBACs are transported (and how) or installed by the CVO**
+1. The Rebase Bot then proceeds creating a Pull Request (PR) containing the rebase changes to the main branch and the generated manifests. The PR then gets reviewed and merged into the main branch.
+1. Shortly after, the Provider's repository gets built by the ART builders and ends up being included into the payload. Also the transport ConfigMap(s) end up being embedded in the payload.
+1. Once the payload is installed and a new cluster is created the CVO kicks in and applies the manifests embedded introduced by the CAPI Provider repository and embedded in the payload. This results in the creation of the transport ConfigMap(s) in the APIServer. At this point the CVO also launches the [Cluster CAPI Operator](https://github.com/openshift/cluster-capi-operator).
+1. The Cluster CAPI Operator controllers start up. The _CAPI Installer controller_ loads the `capi-providers-list` ConfigMap from the APIServer (If none exists, a default one gets pre-created by the controller on the basis of what platform is detected in the Infrastructure Object's status).
+This ConfigMap is constantly watched also at later stages of the controller execution, and it is reacted upon changes.  **TODO(@damdo): define the format of the ConfigMap (in API Extensions?)**
+1. In Parallel the other Cluster CAPI Operator controllers perform their usual duties (Syncing of the user-data-secret from MAPI to CAPI namespace, Kubeconfig generation fro CAPI, Setting Infrastructure Cluster to be externally managed and ready, setting of the `ControlPlaneInitializedCondition` to `true`)
+1. The Cluster CAPI Operator's _CAPI Installer controller_ fetches the transport ConfigMap(s) according to what is defined in the `capi-providers-list`, extracts the manifests from the ConfigMap(s) and applies them to the APIServer.
+These transport ConfigMaps are constantly watched, also at later stages of the controller execution, and the controller reacts upon changes, keeping the desired state in the ConfigMaps in sync with the actual state in the cluster.
+1. The Cluster API Provider controllers start up at this point, and are ready to accept CAPI specific CRs (Cluster, InfraCluster, MachineDeployment, MachineSet, ...)
+
+
+### API Extensions
+N/A
```

Comment:
Should define how the CAPI transport configmaps will be identified, what is the contract between the operator and the operands here

Reply:
+1

Reply:
Added details on what contracts we will rely on.